File: tests/fixtures/go_lane/recordparser/parser.go
// SPDX-License-Identifier: Apache-2.0
// A tiny untrusted-input parser fixture for the govfuzz Go lane. ParseRecord is
// the fuzzable entry point; a crafted first byte drives an out-of-bounds slice
// access (CWE-125), a real panic reachable purely from the input bytes.
package recordparser

// ParseRecord parses a length-prefixed record. A planted bug: on tag 'A' it reads
// past the slice bounds without checking length.
func ParseRecord(data []byte) int {
	if len(data) == 0 {
		return 0
	}
	tag := data[0]
	if tag == 'A' {
		// Bug: assumes at least 5 bytes follow the tag; panics (index out of range)
		// on a short input.
		return int(data[1]) + int(data[2]) + int(data[3]) + int(data[4]) + int(data[5])
	}
	return len(data)
}

File: tests/fixtures/c_private_handle/parser.c
/* SPDX-License-Identifier: Apache-2.0 */
#include "parser.h"

/* Private: complete ONLY in this translation unit. */
struct pv_session {
    int limit;
    unsigned char seen[4];
};

int pv_session_scan(pv_session *s, const unsigned char *data, size_t len) {
    if (s == NULL || data == NULL || len < 2) {
        return 0;
    }
    if (data[0] == 'G') {
        /* Planted out-of-bounds read (GF-201/CWE-125): `seen` holds 4 bytes. */
        return s->seen[len];
    }
    return (int)len;
}
